Do not prevent logging out when the main window is visible.
--- a/src/container.cpp
+++ b/src/container.cpp
@@ -78,6 +78,7 @@
 
 Container::Container()
         : QMainWindow(),
+          m_quitOk( false ),
           m_userCheck( false ),
           m_soundcardError( false ),
           m_sidebarEnabled( false ),
@@ -575,7 +576,7 @@
 void
 Container::closeEvent( QCloseEvent *event )
 {
-    bool quit = false;
+    bool quit = m_quitOk;
 
     #ifdef Q_WS_MAC
     if ( !event->spontaneous() )
@@ -610,6 +611,8 @@
         minimiseToTray();
         event->ignore();
     }
+
+    m_quitOk = false;
 }
 
 
--- a/src/container.h
+++ b/src/container.h
@@ -48,6 +48,7 @@
         static Container& instance() { return *s_instance; }
         class ShareDialog& shareDialog() { return *m_shareDialog; }
         int stackIndex() const { return ui.stack->currentIndex(); }
+        void setQuitOk( bool quitOk ) { m_quitOk = quitOk; }
 
         std::vector<class CPluginInfo>& getPluginList();
 
@@ -108,6 +109,7 @@
         class ShareDialog *m_shareDialog;
         class DiagnosticsDialog *m_diagnosticsDialog;
 
+        bool m_quitOk;
         bool m_userCheck;
         bool m_soundcardError;
         bool m_sidebarEnabled;
--- a/src/lastfmapplication.cpp
+++ b/src/lastfmapplication.cpp
@@ -368,6 +368,14 @@
 
 
 void
+LastFmApplication::commitData( QSessionManager & sm )
+{
+    m_container->setQuitOk( true );
+    QApplication::commitData( sm );
+}
+
+
+void
 LastFmApplication::initTranslator()
 {
     QString langCode;
--- a/src/lastfmapplication.h
+++ b/src/lastfmapplication.h
@@ -139,6 +139,7 @@
 public:
     LastFmApplication( int& argc, char** argv );
     ~LastFmApplication();
+    void commitData( QSessionManager & sm );
 
     void setUser( const QString& username );
 
